app/src/hooks/useFees.ts

```diff
@@ -637,3 +591,11 @@ const checkEip1559BalanceSufficiency = async (
 
   return erc20Balance >= parseUnits(sourceTokenAmount.toString(), sourceToken.decimals)
 }
+
+function getToken(location: TLocation | undefined, symbol: string) {
+  const token = location ? getTokenByLocation(location) : getTokenFromSymbol(symbol)
+  if (!token) {
+    throw new Error(`Token not found for location: ${JSON.stringify(location)}`)
```

Comment:
Nice function. The error here could be more precise tho. 
```suggestion
  const token = location ? getTokenByLocation(location) : getTokenFromSymbol(symbol)
  if (!token) {
    throw new Error(`Token not found for location ${JSON.stringify(location)} and symbol ${symbol}`)
```